ethdb/privateapi/ethbackend.go

```diff
@@ -257,11 +265,14 @@ func (s *EthBackendServer) EngineExecutePayloadV1(ctx context.Context, req *type
 	}
 
 	executedStatus := <-s.statusCh
-
+	// Discard all previous prepared payloads if another block was proposed
 	if executedStatus.Error != nil {
 		return nil, executedStatus.Error
 	}
-
+	// Discard all payload assembled
+	atomic.StoreUint32(s.pauseAssemble, 1)
+	s.pendingPayloads = make(map[uint64]types2.ExecutionPayload)
```

Comment:
Probably should be protected by `mu`.

Reply:
Also, it's probably better to protect the entire `EngineExecutePayloadV1` with
```
	s.mu.Lock()
	defer s.mu.Unlock()
```
in the beginning so that we prevent its concurrent execution with `EngineGetPayloadV1` & `EngineForkChoiceUpdatedV1`.